lib/src/upstream.rs

```diff
@@ -115,17 +116,27 @@ impl hyper::service::Service<Uri> for BackendConnector {
         // the future for establishing the TCP connection. we create this outside of the `async`
         // block to avoid capturing `http`
         let connect_fut = self.http.call(backend.uri.clone());
+        let mut custom_roots = self.tls_config.roots.clone();
+        let (added, ignored) = custom_roots.add_parsable_certificates(&self.backend.ca_certs);
```

Comment:
I don't believe we should be adding this cert to the roots, but rather using it in isolation.

Reply:
what isolation are you looking for? `custom_roots` is a copy of the root cert store for this invocation of the `Connector`, so the additional `self.backend.ca_roots` is only present in the custom root CA set provided for this specific connection attempt

Reply:
(i'd misunderstood the original comment, "isolation" here means "if there's a CA cert, it should be the only cert used in a root store for validation" rather than the behavior in this PR of "use the CA cert, and the system roots, as acceptable roots for the remote certificate")

Reply:
IOW, if we cannot parse the cert we will just proceed with no cert? Is that correct?

Reply:
That's correct!